readthedocs/builds/models.py

```diff
@@ -879,6 +881,28 @@ def external_version_name(self):
     def using_latest_config(self):
         return int(self.config.get('version', '1')) == LATEST_CONFIGURATION_VERSION
 
+    @property
+    def external_build_title(self):
+        """Returns the title of the pull/merge request by parsing HttpExchange object."""
+        if self.is_external:
+            sha_str = f'"sha": "{self.commit}",'
+            number_str = f'"number": {self.version.verbose_name},'
+            integration_list = self.project.integrations.all().values_list('id', flat=True)
+            http_exchange = HttpExchange.objects.filter(
+                Q(object_id__in=integration_list) &
+                Q(request_body__contains=sha_str) &
+                Q(request_body__contains=number_str)
+            ).first()
```

Comment:
I'm worried about this query. It feels really heavy for a page that is loaded so much. Is there not an easier way to do this? We should cache the data on the build if we want to display it.

Reply:
Maybe its better to add a field where we will store this. It will be cleaner and less complex.

Reply:
I'm :+1: on adding a field in the db. Although, I don't like the idea of executing this query with `contains` on these objects because that data could be really big.

I'm echoing Eric's question here, "Is there not an easier way to do this?"

Reply:
Another Approach can be converting `TextField` to `JSONField` which will make it much easier to query
ref: https://docs.djangoproject.com/en/2.2/ref/contrib/postgres/fields/#key-index-and-path-lookups